x/hard/keeper/repay.go

```diff
@@ -43,6 +43,13 @@ func (k Keeper) Repay(ctx sdk.Context, sender sdk.AccAddress, coins sdk.Coins) e
 		return err
 	}
 
+	// If any coin denoms have been completely repaid reset the denom's borrow index factor
+	for _, coin := range payment {
+		if coin.Amount.Equal(borrow.Amount.AmountOf(coin.Denom)) {
+			borrow.Index = borrow.Index.SetInterestFactor(coin.Denom, sdk.ZeroDec())
+		}
```

Comment:
nit: I feel like it's slightly safer to remove the interest factor rather than set to zero.
If any other part of the code assumes the values in `borrow.Index` are correct it could run into issues since zero is not a valid factor value. But open to other opions.

Reply:
Agree that removing the factor is best option. I don't think it has any security implications, but makes the state machine a little more explicit (no open borrow = no factor, as opposed to no open borrow = no factor or zero factor).

Reply:
Updated in https://github.com/Kava-Labs/kava/pull/794/commits/6ece947e2f0885f45b9235402b3644358cb4c6b4